src/Extension/FluentSearchFormExtension.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/**
+ * SilverStripe core search work around when using Fluent
+ * @link https://gist.github.com/baukezwaan/266c469758319daea5460f2a5647c525
+ */
+
+namespace TractorCow\Fluent\Extension;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\CMS\Search\SearchForm;
+use SilverStripe\ORM\ArrayList;
+use SilverStripe\ORM\DB;
+use SilverStripe\ORM\SS_List;
+use TractorCow\Fluent\State\FluentState;
+
+/**
+ * Simple extension to enable SS Core search to work with SS Fluent
+ */
+class FluentSearchFormExtension extends SearchForm
+{
+    /**
+     * Return dataObjectSet of the results using current request to get info from form.
+     * Wraps around {@link searchEngine()}.
+     *
+     * @return SS_List
+     */
+    public function getResults()
+    {
+        // Get request data from request handler
+        $request = $this->getRequestHandler()->getRequest();
+
+        $keywords = $request->requestVar('Search');
+
+        $andProcessor = function ($matches) {
+            return ' +' . $matches[2] . ' +' . $matches[4] . ' ';
+        };
+
+        $notProcessor = function ($matches) {
+            return ' -' . $matches[3];
+        };
+
+        $keyword_patterns = [
+            'andProcessor' => '/()("[^()"]+")( and )("[^"()]+")()/i',
+            'andProcessor' => '/(^| )([^() ]+)( and )([^ ()]+)( |$)/i',
+            'notProcessor' => '/(^| )(not )("[^"()]+")/i',
+            'notProcessor' => '/(^| )(not )([^() ]+)( |$)/i',
+        ];
+
+        foreach ($keyword_patterns as $callback => $pattern) {
+            $keywords = preg_replace_callback($pattern, $$callback, $keywords);
+        }
+
+        $keywords = DB::get_conn()
+            ->escapeString(
```

Comment:
With parameterised queries don't escapeString(). You must pass it in without escape.

Reply:
What would you use in place of `escapeString()`? Would `quoteString()` suffice?

Reply:
I wouldn't use either. By using prepared_query(), you are delegating the escaping of this string to the internal SQL engine. :)

```
$keywords = $request->requestVar('Search');
```

What you've put is exactly what you want.